UM/Qt/qml/UM/CheckBox.qml

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2021 Ultimaker B.V.
+// Uranium is released under the terms of the LGPLv3 or higher.
+
+import QtQuick 2.10
+import QtQuick.Controls 2.3
+
+import UM 1.3 as UM
+
+CheckBox
+{
+    id: control
+
+    hoverEnabled: true
+
+    indicator: Rectangle
+    {
+        height: UM.Theme.getSize("checkbox").height
+        width: UM.Theme.getSize("checkbox").width
+
+        anchors.verticalCenter: parent.verticalCenter
+
+        color:
+        {
+            if (!control.enabled)
+            {
+                return UM.Theme.getColor("checkbox_disabled")
+            }
+            return UM.Theme.getColor("checkbox")
+        }
+
+        radius: UM.Theme.getSize("checkbox_radius").width
+        border.width: UM.Theme.getSize("default_lining").width
+        border.color:
+        {
+            if (!enabled)
+            {
+                return UM.Theme.getColor("checkbox_border")
+            }
+            if (control.hovered || control.activeFocus)
+            {
+                return UM.Theme.getColor("checkbox_border_hover")
+            }
+            return UM.Theme.getColor("checkbox_border")
+        }
+
+        UM.RecolorImage
+        {
+            anchors.verticalCenter: parent.verticalCenter
+            anchors.horizontalCenter: parent.horizontalCenter
+
+            height:
+            {
+                switch(control.checkState)
```

Comment:
This is generally fine, but for the future you might want to have a look at states: https://doc.qt.io/qt-5/qml-qtquick-state.html

This will have the exact same result, but what i prefer about it is that you define all the changes that need to happen to a component given a certain state (eg; the check state) in a single place.